Libraries/PushNotificationIOS/RCTPushNotificationManager.m

```diff
@@ -35,6 +35,8 @@ + (UILocalNotification *)UILocalNotification:(id)json
   UILocalNotification *notification = [UILocalNotification new];
   notification.fireDate = [RCTConvert NSDate:details[@"fireDate"]] ?: [NSDate date];
   notification.alertBody = [RCTConvert NSString:details[@"alertBody"]];
+  notification.alertAction = [RCTConvert NSString:details[@"alertAction"]] ?: nil;
+  notification.applicationIconBadgeNumber = [RCTConvert NSInteger:details[@"applicationIconBadgeNumber"]] ?: 0;
```

Comment:
Are you sure that you want to _clear_ the badge number if the entry was not included in the payload? That would make it effectively impossible to not touch the badge number with a payload.